pkg/loadbalance/smooth_weighted_round_robin.go

```diff
@@ -0,0 +1,165 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package loadbalance
+
+import (
+	"errors"
+)
+
+type weightedItem struct {
+	Item            interface{}
+	Weight          float64
+	CurrentWeight   float64
+	EffectiveWeight float64
+}
+
+// SmoothWeightedRR - Load Balancer implementation, Smooth Weighted Round Robin
+type SmoothWeightedRR struct {
```

Comment:
Keep the implementation private.

```suggestion
type smoothWeightedRR struct {
```